EventFilter/L1TRawToDigi/src/implementations_stage2/EtSumPacker.cc

```diff
@@ -36,30 +36,42 @@ namespace stage2 {
       edm::Handle<EtSumBxCollection> etSums;
       event.getByToken(static_cast<const CommonTokens*>(toks)->getEtSumToken(), etSums);
 
-      uint32_t et_word = 0;
-      uint32_t ht_word = 0;
-      uint32_t met_word = 0;
-      uint32_t mht_word = 0;
-
+      std::vector<uint32_t> load;
+      int nBx = 0;
       for (int i = etSums->getFirstBX(); i <= etSums->getLastBX(); ++i) {
+
+	 uint32_t et_word = 0;
+	 uint32_t ht_word = 0;
+	 uint32_t met_word = 0;
+	 uint32_t mht_word = 0;
+
          for (auto j = etSums->begin(i); j != etSums->end(i); ++j) {
 	   uint32_t word = std::min(j->hwPt(), 0xFFF);
 	   if ((j->getType()==l1t::EtSum::kMissingEt) || (j->getType()==l1t::EtSum::kMissingHt))
 	     word = word | ((j->hwPhi() & 0xFF) << 12);
 	   
-	   if (j->getType()==l1t::EtSum::kTotalEt)   et_word = word;
-	   if (j->getType()==l1t::EtSum::kTotalHt)   ht_word = word;
-	   if (j->getType()==l1t::EtSum::kMissingEt) met_word = word;
-	   if (j->getType()==l1t::EtSum::kMissingHt) mht_word = word;
+	   if (j->getType()==l1t::EtSum::kTotalEt)       et_word  |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFP0)   et_word  |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kTotalHt)       ht_word  |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFM0)   ht_word  |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kMissingEt)     met_word |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFP1)   met_word |= (word << 28);
+	   if (j->getType()==l1t::EtSum::kMissingHt)     mht_word |= word;
+	   if (j->getType()==l1t::EtSum::kMinBiasHFM1)   mht_word |= (word << 28);
```

Comment:
Hi Brian, 

@blwiner

I’ve updated the PR to include the multiBX packing:

https://github.com/cms-l1t-offline/cmssw/pull/374

This PR includes the changes from the previous PR, which adds the new EtSums to EtSumPacker, 
and also includes a fix to the caloTowerHash.

Please let me know if this looks correct.

Cheers,
Aaron

> On 1 Jul 2016, at 18:53, blwiner <notifications@github.com> wrote:
> 
> In EventFilter/L1TRawToDigi/src/implementations_stage2/EtSumPacker.cc <https://github.com/cms-l1t-offline/cmssw/pull/333#discussion_r69327507>:
> 
> >           for (auto j = etSums->begin(i); j != etSums->end(i); ++j) {
> >  	   uint32_t word = std::min(j->hwPt(), 0xFFF);
> >  	   if ((j->getType()==l1t::EtSum::kMissingEt) || (j->getType()==l1t::EtSum::kMissingHt))
> >  	     word = word | ((j->hwPhi() & 0xFF) << 12);
> >  	   
> > -	   if (j->getType()==l1t::EtSum::kTotalEt)   et_word = word;
> > -	   if (j->getType()==l1t::EtSum::kTotalHt)   ht_word = word;
> > -	   if (j->getType()==l1t::EtSum::kMissingEt) met_word = word;
> > -	   if (j->getType()==l1t::EtSum::kMissingHt) mht_word = word;
> > +	   if (j->getType()==l1t::EtSum::kTotalEt)       et_word  |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFP0)   et_word  |= (word << 28);
> > +	   if (j->getType()==l1t::EtSum::kTotalHt)       ht_word  |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFM0)   ht_word  |= (word << 28);
> > +	   if (j->getType()==l1t::EtSum::kMissingEt)     met_word |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFP1)   met_word |= (word << 28);
> > +	   if (j->getType()==l1t::EtSum::kMissingHt)     mht_word |= word;
> > +	   if (j->getType()==l1t::EtSum::kMinBiasHFM1)   mht_word |= (word << 28);
> @bundocka <https://github.com/bundocka> Aaron, I think what you have changed takes care of the new EtSums, etc. However, it is not clear to me that it fixes the problem with packing of multiBX events. Please see #333 <https://github.com/cms-l1t-offline/cmssw/pull/333> Where I proposed some changes. Specifically, the old code added zero padding at the end for empty words. But this must be done for each BX to get the proper format. -Brian
> … <x-msg://18/#>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
> 
>  <https://github.com/cms-l1t-offline/cmssw/pull/333#discussion_r69200419> <https://github.com/cms-l1t-offline/cmssw/pull/371> <https://github.com/cms-l1t-offline/cmssw/pull/371> <https://github.com/cms-l1t-offline/cmssw/pull/371> <https://github.com/cms-l1t-offline/cmssw/pull/371> <https://github.com/cms-l1t-offline/cmssw/pull/333> <https://github.com/cms-l1t-offline/cmssw/pull/333#discussion_r69142948> <https://github.com/jimbrooke> <https://github.com/jimbrooke> <https://github.com/jimbrooke> <https://github.com/bundocka> <https://github.com/bundocka> <https://github.com/bundocka> <http://etsumpacker.cc/> <https://github.com/cms-l1t-offline/cmssw/pull/333> <https://github.com/cms-l1t-offline/cmssw/pull/333#discussion_r69123226> <https://github.com/cms-l1t-offline/cmssw/pull/333/files/9d205b3c737d05eaebd4b301c6fe7750570b2117#r69123226> <https://github.com/notifications/unsubscribe/AElaBv9issVRE23kdJzLPmPO1EHfk9b4ks5qQ7d9gaJpZM4IlnuI> <https://github.com/cms-l1t-offline/cmssw/pull/333/files/9d205b3c737d05eaebd4b301c6fe7750570b2117#r69142948> <https://github.com/notifications/unsubscribe/AMqRT4LEYZm6RKLbOvcCbU4TEDVV581rks5qQ9NogaJpZM4IlnuI> <https://github.com/cms-l1t-offline/cmssw/pull/333/files/9d205b3c737d05eaebd4b301c6fe7750570b2117#r69200419> <https://github.com/notifications/unsubscribe/AElaBrG0O8VQ5gCwDft5h11rKo4NFc8sks5qRCEHgaJpZM4IlnuI>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/cms-l1t-offline/cmssw/pull/333/files/9d205b3c737d05eaebd4b301c6fe7750570b2117#r69327507>, or mute the thread <https://github.com/notifications/unsubscribe/AMqRT1SOUe3__PG5e8moX90f6QGr4_tzks5qRUXwgaJpZM4IlnuI>.
>